app/src/main/java/com/andrew/tasky/core/MainActivity.kt

```diff
@@ -1,15 +1,31 @@
 package com.andrew.tasky.core
 
+import android.content.SharedPreferences
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
+import androidx.navigation.NavController
+import androidx.navigation.fragment.NavHostFragment
 import com.andrew.tasky.R
+import com.andrew.tasky.core.data.PrefsKeys
 import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
 
 @AndroidEntryPoint
 class MainActivity : AppCompatActivity() {
+    private lateinit var navController: NavController
+    @Inject lateinit var prefs: SharedPreferences
```

Comment:
I'd avoid injecting stuff directly in the UI, I usually try to keep it as dumb as possible and only let the VM decide what the UI should do or display. So reading from prefs could happen in the VM here :)